### PR TITLE
Filter groups and posts based on visibility and owner

### PR DESCRIPTION
**Pull request by artur-gajewski**

### CODE REVIEW

File: src/Cobase/AppBundle/Service/UserService.php [hunk 1]
```diff
@@ -99,4 +99,54 @@ public function getUserByUsername($username)
...
+     * Filter out groups if group is non-public and user is not the owner of the group
+     *
+     * @param $userGroups
+     * @return array
+     */
+    public function filterUnlistedGroups($userGroups, User $user = null)
+    {
+        $groups = array();
+
+        foreach($userGroups as $group) {
```
Comment (stiggg) [inline src/Cobase/AppBundle/Service/UserService.php:112]: NOTE: I'd consider creating a custom DQL query for this kind of filtering, with maybe generalizing the use case with parameterizing the query...

Comment (stiggg) [inline src/Cobase/AppBundle/Service/UserService.php:112]: 1 foreach + 3 if's is a bit too much...

File: src/Cobase/AppBundle/Service/UserService.php [hunk 2]
```diff
@@ -99,4 +99,54 @@ public function getUserByUsername($username)
         );
     }
 
+    /**
+     * Filter out groups if group is non-public and user is not the owner of the group
+     *
+     * @param $userGroups
+     * @return array
+     */
+    public function filterUnlistedGroups($userGroups, User $user = null)
```
Comment (stiggg) [inline src/Cobase/AppBundle/Service/UserService.php]: NOTE: I think you can typehint $userGroups with \ Traversable

File: src/Cobase/AppBundle/Service/UserService.php [hunk 3]
```diff
@@ -99,4 +99,54 @@ public function getUserByUsername($username)
...
+     * @param $userPosts
+     * @return array
+     */
+    public function filterUnlistedGroupPosts($userPosts, User $user = null)
+    {
+        $posts = array();
+
+        foreach($userPosts as $post) {
+            if ($post instanceof Post) {
+                if ($post->getGroup()->getIsPublic() === false) {
```
Comment (stiggg) [inline src/Cobase/AppBundle/Service/UserService.php:139]: BLOCKER: Is it given every Post has Group, or should check for existence and error handle accordingly?

Comment (artur-gajewski) [inline src/Cobase/AppBundle/Service/UserService.php:139]: I would say it is given, logic doesn't make it possible to create one without a group.

**Pull request merged**